packages/server/src/bandit/banditData.ts

```diff
@@ -64,54 +64,65 @@ export interface BanditData {
     bestVariants: BanditVariantData[]; // will contain more than 1 variant if there is a tie
 }
 
-async function buildBanditDataForTest(epicTest: EpicTest): Promise<BanditData> {
-    const samples = await getBanditSamplesForTest(epicTest.name);
-
-    if (samples.length === 0) {
-        // No samples yet, set all means to zero to allow random selection
-        return {
-            testName: epicTest.name,
-            bestVariants: epicTest.variants.map((variant) => ({
-                variantName: variant.name,
-                mean: 0,
-            })),
-        };
-    }
+function getDefaultWeighting(epicTest: EpicTest): BanditData {
+    // No samples yet, set all means to zero to allow random selection
+    return {
+        testName: epicTest.name,
+        bestVariants: epicTest.variants.map((variant) => ({
+            variantName: variant.name,
+            mean: 0,
+        })),
+    };
+}
 
-    const variantsData = samples.flatMap((sample) => sample.variants);
+function calculateMeanPerVariant(samples: TestSample[], epicTest: EpicTest): BanditVariantData[] {
+    const allVariantSamples = samples.flatMap((sample) => sample.variants);
     const variantNames = epicTest.variants.map((variant) => variant.name);
 
-    const variantMeans = variantNames.map((variantName) => {
-        const variantSamples = variantsData.filter(
+    return variantNames.map((variantName) => {
+        const variantSamples = allVariantSamples.filter(
             (variantSample) => variantSample.variantName === variantName,
         );
 
-        const mean = sampleMean(variantSamples);
+        const mean = calculateWeightedMeanOfSamples(variantSamples);
```

Comment:
Is it weighted or just the natural mean? I don't think we're using weights in this version.

Reply:
I think technically they're weighted by the sample size of each mean (ie views per variant per hour) 😵

Reply:
```suggestion
        const mean = calculateOverallMeanOfSamples(variantSamples);
```
better name?

Reply:
Mean is calculated by normalising the sum over the views per variant per hour. Similar but not the same as weighting. We can get into that when we get to weighted selection methods 😀

Reply:
The suggested change is good